libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -189,8 +189,10 @@ bool AC_AutoTune::init_internals(bool _use_poshold,
 // stop - should be called when the ch7/ch8 switch is switched OFF
 void AC_AutoTune::stop()
 {
-    // set gains to their original values
-    load_gains(GAIN_ORIGINAL);
+    if (motors->armed()) {
```

Comment:
Txs for this fix.  I worry a bit about this motors->armed() check being here because it hides the already confusing save/restore behaviour down a level.  Sorry I don't immediately have a suggestion on how it should be done but I worry that the logic that decides whether the vehicle should fly with original or new gains is dispersed over the code.

Reply:
I guess my point is that I'm concerned that the stop is the function that controls what the vehicle could fly with next instead of the start() or init().

Reply:
Problem is we have to load the original gains while flying after an autotune, so I think it needs to be in stop. I could move it to mode_autotune, but I had thought that this class was separated out so it could be shared by copter and plane?

Reply:
@lthall pointed out that this should be redundant, so I have removed it. I am still seeing a bug, just not sure where.